clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -406,6 +406,10 @@ public void close() {
         } finally {
             super.close();
         }
+        Node coordinator;
+        while ((coordinator = coordinator()) != null && client.pendingRequestCount(coordinator) > 0) {
```

Comment:
I think you could use `client.awaitPendingRequests()` to achieve the same thing. I'm wondering if we should have a hard-coded timeout. In the worst case, we'll only block as long as the request timeout, but since both the OffsetCommit and LeaveGroup are "best effort" and should be handled relatively quickly, perhaps it would be better to timeout sooner. What do you think?